src/Data/Primitive/Contiguous/Class.hs

```diff
@@ -872,15 +877,19 @@ instance Contiguous SmallArray where
   {-# INLINE copyMut_ #-}
   copyMut_ = copySmallMutableArray
   {-# INLINE replicateMut #-}
-  replicateMut = replicateSmallMutableArray
+  replicateMut = newSmallArray
   {-# INLINE run #-}
   run = runSmallArrayST
+  {-# INLINE shrink #-}
+  shrink !arr !n = do
+    shrinkSmallMutableArray arr n
+    pure arr
```

Comment:
Excellent! I'd do these 4 lines for PrimArray also, using `shrinkMutablePrimArray`, thanks.

Reply:
Good idea. I've done this in 95e48d02855bcd8eaf7e55d4340f3f138fc7a35e.